server/fishtest/views.py

```diff
@@ -529,11 +529,9 @@ def users_monthly(request):
     return {"users": users_list}
 
 
-def get_master_info():
+def get_master_info(url):
```

Comment:
```python
def get_master_info(url):
    try:
        commits = requests.get(url)
        commits.raise_for_status()
    except Exception as e:
        print(f"Exception getting commits:\n{e}")
        return None
    bench_search = re.compile(r"(^|\s)[Bb]ench[ :]+([1-9]\d{5,9})(?!\d)")
    for idx, commit in enumerate(commits.json()):
        message_lines = commit["commit"]["message"].strip().split("\n")
        bench = bench_search.search(message_lines[-1].strip())
        if idx == 0:
```

Reply:
```diff
    bench_search = re.compile(r"(^|\s)[Bb]ench[ :]+([1-9]\d{5,9})(?!\d)")
    bench_search = re.compile(r"(^|\s)[Bb]ench[ :]+([1-9]\d{5,7})(?!\d)")
```

Reply:
We made Stockfish CI to match to at max 10 digits I think it's a bit late for returning to 8.

Reply:
SF CI code is still broken, try this
"Fix bench

With old code produced bench + 1, at example for this commit we had
bench 123456

close 00000
bench 123457
"

Reply:
> 

Yeah log could be splitted by each commit and each commit stripped to from ending white spaces and only the last line checjed for bench afterwards.

That's the fishtest way of doing it, I just thought it wasn't worth the complexity added for this case.

Reply:
in any ways, I will change it back to original Joost's suggestion and now yours (to a max of 7.)
Feel free to overwrite me on that CI patch in Stockfish repo.
Hoping we haven't bothered Sf devs that much.

Reply:
max of 8

Reply:
```yaml
- name: Extract the bench number from the commit history
  run: |
    N=0
    while true; do
        commit_message=$(git log HEAD -n 1 --skip $N --pretty=format:%B)
        if [[ -z "$commit_message" ]]; then
            break
        fi
        bench_value=$(echo "$commit_message" | tail -n 1 | grep -oP "\b[Bb]ench[ :]\+\K[1-9][0-9]{5,7}\b")
        if [[ -n "$bench_value" ]]; then
            echo "benchref=$bench_value" >> $GITHUB_ENV
            echo "Reference bench: $bench_value"
            break
        fi
        N=$((N+1))
    done
    if [[ -z "$bench_value" ]]; then
        echo "No bench found"
    fi
```